pcap_linux.go

```diff
@@ -108,15 +110,40 @@ func (h *Handle) ReadPacketData() (data []byte, ci gopacket.CaptureInfo, err err
 
 func (h *Handle) readPacketDataSyscall() (data []byte, ci gopacket.CaptureInfo, err error) {
 	b := make([]byte, h.snaplen)
-	read, _, err := syscall.Recvfrom(h.fd, b, 0)
+	oob := make([]byte, syscall.CmsgSpace(tpacketAuxdataSize))
+	n, _, _, _, err := syscall.Recvmsg(h.fd, b, oob, 0)
 	if err != nil {
-		return nil, ci, fmt.Errorf("error reading: %v", err)
+		return nil, ci, fmt.Errorf("error reading packets: %w", err)
+	}
+
+	var auxData syscall.TpacketAuxdata
```

Comment:
I think you may want to pull this to a separate function, especially once you have the mmap version. I assume it will have the same structure? Maybe not?

Reply:
You don't want to know how long it took me to figure it out when I wrote this. Actually, I don't want to know either! 😁